### PR TITLE
VS-93: Address Handling of Nullables

### PR DESCRIPTION
**Pull request by Ravi-Raghavan**
Address Handling of Nullables

### CODE REVIEW

File: src/MongoDB.Analyzer.Helpers/Builders/MqlGenerator.cs [hunk 1]
```diff
@@ -23,8 +24,9 @@ public static class MqlGenerator
     {
 #pragma warning disable CS0169 // These fields are never used, they are needed to ensure that the relevant usings are not accidently removed
 #pragma warning disable IDE0051
-        private static readonly BsonType s_dummyRef1;
-        private static readonly TimeSpanUnits s_dummyRef2;
+        private static readonly Int32 s_dummyRef1; // using System
```
Comment (sanych-sun) [inline src/MongoDB.Analyzer.Helpers/Builders/MqlGenerator.cs:27]: Should we use C# `int` alias instead?

File: src/MongoDB.Analyzer/Core/TypesProcessor.cs [hunk 2]
```diff
@@ -283,7 +283,11 @@ private TypeSyntax CreateTypeSyntaxForSymbol(ITypeSymbol typeSymbol)
         }
         else
         {
-            result = SyntaxFactory.ParseTypeName(ProcessTypeSymbol(typeSymbol));
+            var (isNullable, nonNullableTypeSymbol) = typeSymbol.DiscardNullable();
```
Comment (sanych-sun) [inline src/MongoDB.Analyzer/Core/TypesProcessor.cs:286]: IMHO: `underlingTypeSymbol` sounds better then `nonNullableTypeSymbol`.

File: src/MongoDB.Analyzer/Core/Utilities/SymbolExtensions.cs [hunk 3]
```diff
@@ -69,6 +69,13 @@ internal static class SymbolExtensions
         "System.Type"
     };
 
+    public static (bool IsNullable, ITypeSymbol underlyingType) DiscardNullable(this ITypeSymbol typeSymbol) =>
+        typeSymbol?.OriginalDefinition.SpecialType switch
+        {
+            SpecialType.System_Nullable_T => (true, ((INamedTypeSymbol)typeSymbol).TypeArguments.FirstOrDefault()),
```
Comment (sanych-sun) [inline src/MongoDB.Analyzer/Core/Utilities/SymbolExtensions.cs:75]: Should it be `SingleOrDefault`?

**Review by sanych-sun — APPROVED** [review]
LGTM + minor comments

File: src/MongoDB.Analyzer.Helpers/Builders/MqlGenerator.cs (hunk 1 above)
Comment (BorisDog) [reply src/MongoDB.Analyzer.Helpers/Builders/MqlGenerator.cs:27]: Those dummy variable are used to capture the usings, so they are not removed by mistake.
`Int32 `is used for `System`, while `int `does not require `System`.

File: src/MongoDB.Analyzer/Core/TypesProcessor.cs (hunk 2 above)
Comment (BorisDog) [reply src/MongoDB.Analyzer/Core/TypesProcessor.cs]: Done.

File: src/MongoDB.Analyzer/Core/Utilities/SymbolExtensions.cs (hunk 3 above)
Comment (BorisDog) [reply src/MongoDB.Analyzer/Core/Utilities/SymbolExtensions.cs]: Yes, thanks.

**Review by BorisDog — APPROVED** [review]

**Pull request merged**